Userland/Applications/Screenshot/ScreenshotWidget.cpp

```diff
@@ -0,0 +1,215 @@
+    m_cancel_button = *find_descendant_of_type_named<GUI::Button>("cancel_button");
+    m_ok_button = *find_descendant_of_type_named<GUI::Button>("ok_button");
+
+    Config::pledge_domains("Screenshot");
+
+    String type = Config::read_string("Screenshot", "General", "ScreenshotType", "Whole");
+    if (type == "Custom") {
+        m_whole_button->set_checked(false);
+        m_custom_button->set_checked(true);
+    } else {
+        m_whole_button->set_checked(true);
+        m_custom_button->set_checked(false);
+    }
+
+    m_delay_spinbox->set_value(Config::read_i32("Screenshot", "General", "Delay", 0));
+    m_copy_checkbox->set_checked(Config::read_bool("Screenshot", "General", "CopyToClipboard", true));
+
+    m_cancel_button->on_click = [&](auto) {
+        GUI::Application::the()->quit();
+    };
+
+    m_ok_button->on_click = [&](auto) {
+        String new_type = "Whole";
+        if (m_custom_button->is_checked()) {
+            new_type = "Custom";
+        }
+
+        Config::write_string("Screenshot", "General", "ScreenshotType", new_type);
+        Config::write_i32("Screenshot", "General", "Delay", m_delay_spinbox->value());
+        Config::write_bool("Screenshot", "General", "CopyToClipboard", m_copy_checkbox->is_checked());
+
+        window()->close();
+
+        String output_path = Core::DateTime::now().to_string("screenshot-%Y-%m-%d-%H-%M-%S.png");
+
+        usleep(m_delay_spinbox->value() * 1000);
+
+        save_screenshot(output_path, m_custom_button->is_checked(), m_copy_checkbox->is_checked());
+    };
+}
+
+void ScreenshotWidget::save_screenshot(String output_path, bool selected, bool copy_to_clipboard)
+{
+    Gfx::IntRect crop_region;
+    if (selected) {
+        auto window = GUI::Window::construct();
+        auto& container = window->set_main_widget<SelectableOverlay>(window);
+
+        window->set_title("Screenshot");
+        window->set_has_alpha_channel(true);
+        window->set_fullscreen(true);
+        window->show();
+        GUI::Application::the()->exec();
+
+        crop_region = container.region();
+        if (crop_region.is_empty()) {
+            return;
+        }
+    }
+
+    dbgln("Taking screenshot");
+    Optional<u32> screen_index;
+    auto screen_bitmap = GUI::WindowServerConnection::the().get_screen_bitmap({}, screen_index);
+    dbgln("Taken screenshot");
+
+    RefPtr<Gfx::Bitmap> bitmap = screen_bitmap.bitmap();
+    if (!bitmap) {
+        warnln("Failed to grab screenshot!");
+        return;
+    }
+
+    if (copy_to_clipboard) {
+        GUI::Clipboard::the().set_bitmap(*bitmap);
+
+        auto notification = GUI::Notification::construct();
+        notification->set_title("Screenshot taken");
+        notification->set_text("Screenshot saved to clipboard");
+        notification->show();
+
+        return;
+    }
+
+    if (selected) {
+        bitmap = bitmap->cropped(crop_region);
+    }
```

Comment:
You're copying to the clipboard before cropping the image, even if there's a selection made. Instead of doing it this way, you should just pass in `crop_region` as the first argument for `GUI::WindowServerConnection::the().get_screen_bitmap`

Reply:
I have done it this way since I was having some issues while testing. see ed3b1b0.

Reply:
did you rebase to the new master? If you'll notice the PR that added this change also had a fix in the WindowServer code that passes you the screen bitmap to fix this issue.

Reply:
I did not, that explains why it didn't work for me, I will create a new commit to undo this change.